examples/boundingbox_camera/GlutWindow.cc

```diff
@@ -0,0 +1,450 @@
+          g_target.point);
+      double amount = ((-g_mouse.dragY /
+          static_cast<double>(rayCamera->ImageHeight()))
+          * distance * tan(vfov/2.0) * 6.0);
+      for (ir::CameraPtr camera : g_cameras)
+      {
+        g_viewControl.SetCamera(camera);
+        g_viewControl.SetTarget(g_target.point);
+        g_viewControl.Zoom(amount);
+      }
+    }
+  }
+
+}
+
+//////////////////////////////////////////////////
+void displayCB()
+{
+#if __APPLE__
+  CGLSetCurrentContext(g_context);
+#elif _WIN32
+#else
+  if (g_display)
+  {
+    glXMakeCurrent(g_display, g_drawable, g_context);
+  }
+#endif
+
+  g_cameras[0]->Capture(*g_image);
+  g_cameras[1]->Update();
+  handleMouse();
+
+#if __APPLE__
+  CGLSetCurrentContext(g_glutContext);
+#elif _WIN32
+#else
+  glXMakeCurrent(g_glutDisplay, g_glutDrawable, g_glutContext);
+#endif
+
+  unsigned char *data = g_image->Data<unsigned char>();
+
+  glClearColor(0.5, 0.5, 0.5, 1);
+  glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+  glPixelZoom(1, -1);
+  glRasterPos2f(-1, 1);
+  glDrawPixels(imgw, imgh, GL_RGB, GL_UNSIGNED_BYTE, data);
+
+  glutSwapBuffers();
+}
+
+//////////////////////////////////////////////////
+void idleCB()
+{
+  glutPostRedisplay();
+}
+
+//////////////////////////////////////////////////
+bool SaveImage(const uint8_t *_data)
+{
+  std::string savePath = "save";
+  uint width = g_camera->ImageWidth();
+  uint height = g_camera->ImageHeight();
+
+  // Attempt to create the directory if it doesn't exist
+  if (!ignition::common::isDirectory(savePath))
+  {
+    if (!ignition::common::createDirectories(savePath))
+      return false;
+  }
+
+  std::string filename = "image" + std::to_string(g_counter) + ".png";
+
+  ignition::common::Image localImage;
+  localImage.SetFromData(_data, width, height, ignition::common::Image::RGB_INT8);
+
+  localImage.SavePNG(ignition::common::joinPaths(savePath, filename));
+  return true;
+}
+
+void SaveBoxes(std::vector<ir::BoundingBox> &boxes)
+{
+  std::string savePath = "boxes";
+
+  // Attempt to create the directory if it doesn't exist
+  if (!ignition::common::isDirectory(savePath))
+  {
+    if (!ignition::common::createDirectories(savePath))
+      return;
+  }
+
+  std::string filename = savePath + "/boxes" + std::to_string(g_counter) + ".txt";
+  std::ofstream file(filename);
+
+  for (auto box : boxes)
+  {
+    // box.oreintation.Euler)
+    auto x = std::to_string(box.center.X());
+    auto y = std::to_string(box.center.Y());
+    auto z = std::to_string(box.center.Z());
+
+    auto w = std::to_string(box.size.X());
+    auto h = std::to_string(box.size.Y());
+    auto l = std::to_string(box.size.Z());
+
+    auto roll = std::to_string(box.orientation.Roll());
+    auto pitch = std::to_string(box.orientation.Pitch());
+    auto yaw = std::to_string(box.orientation.Yaw());
+
+    std::string sep = " ";
+    std::string boxString = x + sep + y + sep + z + sep + w + sep + h + sep + l + sep +
```

Comment:
the `math::Vector3d` and `math::Quaterniond` implements the `<<` operator so you should be able to just do ` << box.center << " " << box.size << " " << box.orientation`